applicationset/services/scm_provider/azure_devops_test.go

```diff
@@ -0,0 +1,287 @@
+package scm_provider
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	azureMock "github.com/argoproj/argo-cd/v2/applicationset/services/scm_provider/azure_devops/git/mocks"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	azureGit "github.com/microsoft/azure-devops-go-api/azuredevops/git"
+)
+
+type uuidHolder struct {
+	uuid uuid.UUID
+	err  error
+}
+
+func getUuidHolder() uuidHolder {
+	uuid, err := uuid.NewUUID()
+	return uuidHolder{uuid: uuid, err: err}
+}
+
+func s(input string) *string {
+	return &input
+}
+
+func TestAzureDevopsRepoHasPath(t *testing.T) {
+	organization := "myorg"
+	teamProject := "myorg_project"
+	repoName := "myorg_project_repo"
+	path := "dir/subdir/item.yaml"
+	branchName := "my/featurebranch"
+
+	ctx := context.TODO()
+	uuidHolder := getUuidHolder()
```

Comment:
I think you could just use `uuid.New().String()`. It'll panic on error, but that's fine for a test.